framework/generated/vulkan_generators/vulkan_replay_consumer_body_generator.py

```diff
@@ -893,14 +893,14 @@ def make_body_expressions(self, return_type, name, values, is_override):
                                     if need_temp_value:
                                         if value.base_type in self.structs_with_handle_ptrs:
                                             preexpr.append(
-                                                'SetStructHandleLengths<Decoded_{}>({paramname}->GetMetaStructPointer(), {paramname}->GetLength());'
+                                                'SetStructArrayHandleLengths<Decoded_{}>({paramname}->GetMetaStructPointer(), {paramname}->GetLength());'
                                                 .format(
                                                     value.base_type,
                                                     paramname=value.name
                                                 )
                                             )
                                         postexpr.append(
-                                            'AddStructHandles<Decoded_{basetype}>({}, {name}->GetMetaStructPointer(), {}, &GetObjectInfoTable());'
+                                            'AddStructArrayHandles<Decoded_{basetype}>({}, {name}->GetMetaStructPointer(), {name}->GetLength(), {}, {name}->GetLength(), &GetObjectInfoTable());'
```

Comment:
The control flow for this block is:    
```
if value.is_pointer or value.is_array:
    if is_input:
    else:
        if value.is_array:
        else:
            if is_extenal_object:
            elif self.is_handle(value.base_type):
            else:
                if self.is_array_len(value.name, values):
                elif self.is_struct(value.base_type):
                    if value.base_type in self.structs_with_handles:
```

So this is generating the code to consume a pointer to a single (non-array) struct, and that struct contains handles. In that case I wonder if it these should be:
`SetStructHandleLengths({paramname}->GetMetaStructPointer());`
and
`AddStructHandles({}, {name}->GetMetaStructPointer(), {}, &GetObjectInfoTable());`

This avoids the call to `SetStructArrayHandleLengths` which in turn calls `SetStructHandleLengths`.  That being said, I believe this code works correctly as-is so I consider this an optional change, especially if it has already been tested.